src/wallet/walletutil.cpp

```diff
@@ -31,6 +31,8 @@ fs::path GetWalletDir()
 
 static bool IsBerkeleyBtree(const fs::path& path)
 {
+    if (!fs::exists(path)) return false;
```

Comment:
I don't understand this fix. When is this condition expected? Why not just drop the log statement entirely if we don't care about these errors?

Reply:
If this behavior actually does make sense, a more readable and efficient way to implement it would be to drop this new access and change:

```diff
-if (ec) LogPrintf(...)
+if (ec && ec != errc::no_such_file_or_directory) LogPrintf(...)
```

below

Reply:
It happens here:

https://github.com/bitcoin/bitcoin/blob/8da1aa471eb9f062c99c3134d6151961619e9da5/src/wallet/walletutil.cpp#L74

when `it->path() == "... /regtest/wallets/database"`.

> If this behavior actually does make sense, a more readable and efficient way to implement it would be to drop this new access and change:

I had like that but though early checking is more clear.

I also had the check in L74:
```cpp
if (it->status().type() == fs::directory_file && fs::exists(it->path() / "wallet.dat") && IsBerkeleyBtree(it->path() / "wallet.dat")) {
```

But figure out `IsBerkeleyBtree` could early return `false` when the path doesn't exists.